config/initializers/doorkeeper.rb

```diff
@@ -0,0 +1,43 @@
+Doorkeeper.configure do
+  orm :active_record
+
+  # This block will be called to check whether the resource owner is authenticated or not.
+  # called in the AuthorizedApplicationsController by `before_action :authenticate_resource_owner!`
+  resource_owner_authenticator do
+    clearance_session = env[:clearance] # session = Clearance::Session.new(env)
+    @user = clearance_session && clearance_session.current_user
+
+    if @user
+      @user
+    else
+      session[:return_to] = request.fullpath
+      redirect_to(sign_in_url)
+    end
+  end
+
+  # Restrict access to the web interface for adding oauth authorized applications, you need to declare the block below.
+  # called in the ApplicationsController by `before_action :authenticate_admin!`
+  admin_authenticator do
+    authenticate_resource_owner!
+  end
```

Comment:
both oauth/authorized_applications and oauth/applications require the web user to be logged in to rubygems.org